testing/test_nose.py

```diff
@@ -377,15 +377,48 @@ def test_io(self):
     result.stdout.fnmatch_lines(["* 1 skipped *"])
 
 
-def test_issue_6517(testdir):
+def test_raises(testdir):
     testdir.makepyfile(
         """
         from nose.tools import raises
 
         @raises(RuntimeError)
-        def test_fail_without_tcp():
+        def test_raises_runtimeerror():
             raise RuntimeError
+
+        @raises(Exception)
+        def test_raises_baseexception_not_caught():
+            raise BaseException
+
+        @raises(BaseException)
+        def test_raises_baseexception_caught():
+            raise BaseException
         """
     )
-    result = testdir.runpytest()
-    result.stdout.fnmatch_lines(["* 1 passed *"])
+    result = testdir.runpytest("-vv")
+    result.stdout.fnmatch_lines(
+        [
+            "test_raises.py::test_raises_runtimeerror PASSED*",
+            "test_raises.py::test_raises_baseexception_not_caught FAILED*",
+            "test_raises.py::test_raises_baseexception_caught PASSED*",
```

Comment:
Without the cast here it would display "test_raises.py::test_raises_runtimeerror <- /…/test_raises0/test_raises.py PASSED` here.

Reply:
Hmm sorry, what "cast" you mean?

Reply:
@nicoddemus https://github.com/blueyed/pytest/blob/156a41a855d28ae9f6d83545af69b51fb26f199e/src/_pytest/terminal.py#L747-L752

Since `location` fails to make the path relative it is absolute, and then triggers that code.